### DISCUSSION TITLE
Using the Visual Studio extension in a protected environment

### DISCUSSION THREAD

**Discussion by AnReZa** [microsoft/typespec#3362] [Q&A]
I would like to use the VS extension. Unfortunately, the execution of code in the AppData\Roaming\NPM directory is disabled by policy. I get the message that the execution of TSP-SERVER.CMD is blocked. How can I move this file to another, authorized directory?

**Comment by timotheeguerin** [comment]
The extensions should look up for the locally installed typespec compiler. Do you open a folder that has it installed?

if it’s not in the standard directory(node_modules/@typespec/compiler) you can tell the extension where it is with the config here  https://typespec.io/docs/introduction/editor/vs#configure

**Reply by AnReZa** [reply]
Well, I installed everything following using [this](https://typespec.io/docs) guide and then created a test project in my project folder under `C:\dev\tsp-test\main.tsp`.

**Reply by timotheeguerin** [reply]
Do you have a `node_modules` folder with `@typespec/compiler` in `C:/dev/tsp-test`?

**Reply by AnReZa** [reply]
Yes.

**Reply by timotheeguerin** [reply]
can you share the full error? have you tried setting the config above and restarting visual studio?
can you also confirm that runnnign `npx tsp compile .` is working in that directory
